Framework/DataHandling/src/LoadIsawDetCal.cpp

```diff
@@ -515,5 +392,88 @@ std::vector<std::string> LoadIsawDetCal::getFilenames() {
   return filenamesFromPropertyUnraveld;
 }
 
+/**
+ * Perform the rotation for the calibration
+ *
+ * @param rX the vector of (base_x, base_y, base_z) from the calibration file
+ * @param rY the vector of (up_x, up_y, up_z) from the calibration file
+ * @param detectorInfo the DetectorInfo object from the workspace
+ * @param comp the component to rotate
+ * @param doWishCorrection if true apply a special correction for WISH
+ */
+void LoadIsawDetCal::doRotation(V3D rX, V3D rY, DetectorInfo &detectorInfo,
+                                boost::shared_ptr<const IComponent> comp,
+                                bool doWishCorrection) {
+  // These are the ISAW axes
+  rX.normalize();
+  rY.normalize();
+
+  // These are the original axes
+  const V3D oX(1., 0., 0.);
+  const V3D oY(0., 1., 0.);
+
+  // Axis that rotates X
+  V3D ax1 = oX.cross_prod(rX);
+  // Rotation angle from oX to rX
+  double angle1 = oX.angle(rX) * DegreesPerRadian;
+  if (doWishCorrection)
+    angle1 += 180.0;
+  // Create the first quaternion
+  Quat Q1(angle1, ax1);
+
+  // Now we rotate the original Y using Q1
+  V3D roY = oY;
+  Q1.rotate(roY);
+  // Find the axis that rotates oYr onto rY
+  V3D ax2 = roY.cross_prod(rY);
+  const double angle2 = roY.angle(rY) * DegreesPerRadian;
+  Quat Q2(angle2, ax2);
+
+  // Final = those two rotations in succession; Q1 is done first.
+  Quat Rot = Q2 * Q1;
+
+  // Then find the corresponding relative position
+
+  // TODO: This needs checking. It would be expected that parent->getRot()
+  // should be all that is required to get the absolute rotation. This will
+  // affect WISH, but as there are no tests for this it is difficult to verify.
+  // The absolute rotation, when calling comp->getRotation, is taken as
+  // grandparentRot * parentRot * childrot
```

Comment:
I have been looking through the rotation code again, it seems to be rotating the coordinate system. However, it does not rotate around the origin, but around the position of the parent component. This might be correct since we don't know what is in the files, but it looks weird as well. Maybe add another comment on that?

Or remove also this comment and just drop a message to Vickie, who seems to be the original author?